modules/gdpr_tasks/src/Traversal/RightToBeForgottenEntityTraversal.php

```diff
@@ -0,0 +1,232 @@
+   */
+  protected function processEntity(FieldableEntityInterface $entity, GdprFieldConfigEntity $config, $row_id, array &$results) {
+    $entity_success = TRUE;
+    $entity_type = $entity->getEntityTypeId();
+
+    $fields = $this->entityFieldManager->getFieldDefinitions($entity_type, $entity->bundle());
+    $field_configs = $config->getFieldsForBundle($entity->bundle());
+
+    // Re-load a fresh copy of the entity from storage so we don't
+    // end up modifying any other references to the entity in memory.
+    /* @var \Drupal\Core\Entity\FieldableEntityInterface $entity */
+    $entity = $this->entityTypeManager->getStorage($entity_type)
+      ->loadUnchanged($entity->id());
+
+    foreach ($fields as $field_id => $field_definition) {
+      $field_config = isset($field_configs[$field_id]) ? $field_configs[$field_id] : NULL;
+
+      // If the field is not configured, not enabled,
+      // or not enabled for RTF, then skip it.
+      if ($field_config === NULL
+        || !$field_config->enabled
+        || !in_array($field_config->rtf, ['anonymize', 'remove', 'maybe'])) {
+        continue;
+      }
+
+      $mode = $field_config->rtf;
+      $field = $entity->get($field_id);
+
+      $success = TRUE;
+      $msg = NULL;
+      $anonymizer = '';
+
+      if ($mode == 'anonymize') {
+        list($success, $msg, $anonymizer) = $this->anonymize($field, $field_definition, $field_config);
+      }
+      elseif ($mode == 'remove') {
+        list($success, $msg) = $this->remove($field, $entity);
+      }
+
+      if ($success === TRUE) {
+        $results['log'][] = [
+          'entity_id' => $entity->id(),
+          'entity_type' => $entity_type . '.' . $entity->bundle(),
+          'field_name' => $field->getName(),
+          'action' => $mode,
+          'anonymizer' => $anonymizer,
+        ];
+      }
+      else {
+        // Could not anonymize/remove field. Record to errors list.
+        // Prevent entity from being saved.
+        $entity_success = FALSE;
+        $results['errors'][] = $msg;
+      }
+    }
+
+    if ($entity_success) {
+      $results['successes'][] = $entity;
+    }
+    else {
+      $results['failures'][] = $entity;
+    }
+  }
+
+  /**
+   * Removes the field value.
+   *
+   * @param \Drupal\Core\Field\FieldItemListInterface $field
+   *   The current field to process.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to remove.
+   *
+   * @return array
+   *   First element is success boolean, second element is the error message.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  private function remove(FieldItemListInterface $field, EntityInterface $entity) {
+    try {
+      // If this is the entity's ID, treat the removal as remove the entire
+      // entity.
+      $entity_type = $entity->getEntityType();
+      if ($entity_type->getKey('id') == $field->getName()) {
+        $entity->delete();
```

Comment:
This needs to not happen directly in here as this is outside of the transaction.

Reply:
I've changed this to collect deletions into an array which are now handled inside the transaction by the `Anonymizer`